src/array_value.cpp

```diff
@@ -228,13 +228,60 @@ ValuePtr ArrayValue::map(Env *env, Block *block) {
     return new_array;
 }
 
-// TODO: accept integer and return array
-ValuePtr ArrayValue::first(Env *env) {
-    if (size() > 0) {
+ValuePtr ArrayValue::first(Env *env, ValuePtr n) {
+    auto has_count = n != nullptr;
+
+    if (!has_count) {
+        if (is_empty())
+            return NilValue::the();
+
         return (*this)[0];
-    } else {
-        return NilValue::the();
     }
+
+    ArrayValue* array = new ArrayValue();
+
+    n->assert_type(env, Value::Type::Integer, "Integer");
+    nat_int_t n_value = n->as_integer()->to_nat_int_t();
+
+    if (n_value < 0) {
+        env->raise("ArgumentError", "negative array size");
+        return nullptr;
```

Comment:
Should we return anything here?
I saw raise is `[[noreturn]]` but to me it look weird to have the if not "stop" the code but maybe I'm just not familar enough with the codebase.

Reply:
Yeah it looks weird to not have a return I guess, but it's a common pattern throughout the codebase to just `raise` and have no return, so I'm used to it. No return needed.